content/en/docs/collector/building/_index.md

```diff
@@ -2,10 +2,70 @@
 title: Building custom components
 description: Instructions on how to build your own collector components
 weight: 90
+cSpell:ignore: mdatagen ottl
 ---
 
 The OpenTelemetry Collector can not only be extended by existing components, but
 also by custom components, that you develop and build on your own. Here you will
 find instructions on how to build some of those components. For additional
 details take a look into the documents contained within the
 [opentelemetry-collector-contrib repository](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/README.md).
+
+There are several
+[classes](https://github.com/open-telemetry/opentelemetry-collector/blob/67d37183e6ac9b7180fefc6dc3a55f2a75c12fba/cmd/mdatagen/main.go#L192)
```

Comment:
I would link to our documentation, I don't think a link to mdatagen's internals is the right approach here
```suggestion
[classes](/docs/collector/configuration/#basics)
```

Reply:
I don't think we should link at mdatagen at all. The fact that we use mdatagen is an implementation detail: we do not require for you to use mdatagen to create a new component: you just need a `go.mod` and a properly-typed `NewFactory` function and the builder can take it.